main.tf

```diff
@@ -1 +1,240 @@
+      "logs:DeleteLogStream"
+    ]
+    resources = ["*"]
+  }
+
+  #
+  # Protect S3 Buckets
+  #
+
+  statement {
+    sid    = "ProtectS3Buckets"
+    effect = local.protect_s3_buckets_effect
+    actions = [
+      "s3:DeleteBucket",
+      "s3:DeleteObject",
+      "s3:DeleteObjectVersion",
+    ]
+    resources = var.protect_s3_bucket_resources
+  }
+
+  #
+  # Protect IAM Roles
+  #
+
+  statement {
+    sid    = "ProtectIAMRoles"
+    effect = local.protect_iam_roles_effect
+    actions = [
+      "iam:AttachRolePolicy",
+      "iam:DeleteRole",
+      "iam:DeleteRolePermissionsBoundary",
+      "iam:DeleteRolePolicy",
+      "iam:DetachRolePolicy",
+      "iam:PutRolePermissionsBoundary",
+      "iam:PutRolePolicy",
+      "iam:UpdateAssumeRolePolicy",
+      "iam:UpdateRole",
+      "iam:UpdateRoleDescription"
+    ]
+    resources = var.protect_iam_role_resources
+  }
+
+  #
+  # Restrict Regional Operations
+  #
+
+  statement {
+    sid    = "LimitRegions"
+    effect = local.limit_regions_effect
+
+    # These actions do not operate in a specific region, or only run in
+    # a single region, so we don't want to try restricting them by region.
+    not_actions = [
+      "iam:*",
+      "organizations:*",
+      "route53:*",
+      "budgets:*",
+      "waf:*",
+      "cloudfront:*",
+      "globalaccelerator:*",
+      "importexport:*",
+      "support:*",
+      "sts:*"
+    ]
+
+    resources = ["*"]
+
+    condition {
+      test     = "StringNotEquals"
+      variable = "aws:RequestedRegion"
+      values   = var.allowed_regions
+    }
+  }
+}
+
+resource aws_organizations_policy "combined_policy_block" {
+  count       = var.create_combined_policy_block ? 1 : 0
+  name        = "combined-policy"
+  description = "Combines a given set of policies"
+  content     = data.aws_iam_policy_document.combined_policy_block.json
+
+}
+
+resource "aws_organizations_policy_attachment" "combined_policy_block" {
+  count     = var.create_combined_policy_block ? 1 : 0
+  policy_id = aws_organizations_policy.combined_policy_block[0].id
+  target_id = var.target
+}
+
+locals {
+  deny_root_account_effect             = var.deny_root_account ? "Deny" : "Allow"
+  deny_leaving_orgs_effect             = var.deny_leaving_orgs ? "Deny" : "Allow"
+  deny_creating_iam_users_effect       = var.deny_creating_iam_users ? "Deny" : "Allow"
+  deny_deleting_kms_keys_effect        = var.deny_deleting_kms_keys ? "Deny" : "Allow"
+  deny_deleting_route53_zones_effect   = var.deny_deleting_route53_zones ? "Deny" : "Allow"
+  require_s3_encryption_effect         = var.require_s3_encryption ? "Deny" : "Allow"
+  deny_deleting_cloudwatch_logs_effect = var.deny_deleting_cloudwatch_logs ? "Deny" : "Allow"
+  protect_s3_buckets_effect            = var.protect_s3_buckets ? "Deny" : "Allow"
+  protect_iam_roles_effect             = var.protect_iam_roles ? "Deny" : "Allow"
+  limit_regions_effect                 = var.limit_regions ? "Deny" : "Allow"
+}
```

Comment:
This seems like a weird place for these declarations -- maybe put them at the beginning of the file?